visualization.js

```diff
@@ -0,0 +1,141 @@
+// wait untill page is loaded
+window.document.addEventListener('DOMContentLoaded', load);
+
+function load() {
+	
+    // throttle function
+    function throttle (callback, limit) {
```

Comment:
This function should be defined outside the load function. Nesting functions can be done but is generally a problem because it limits the use of a function to that level of the code, making it less reusable, and less easily findable.